### ISSUE TITLE
Directory Naming Standardization

### ISSUE THREAD

**Issue by thedavidprice** [redwoodjs/redwood#183]
_This is a non-urgent issue that can wait until after March 10th milestone_

See #110 Specifically this comment:
https://github.com/redwoodjs/redwood/pull/110#issuecomment-593838225

I recently ran into an issue where my local clone was still using a folder with a capitalized “G”, while the origin/master had been changed to “g”. (Git doesn’t pay attention to capitalization changes.) While diagnosing, RC suggested we also move to use snake-case instead of camel-case for directories.

Lastly, although there are standards for JS filenames, I couldn’t find anything specific related to directory naming standards.

What say we all?

**Comment by peterp** [comment]
We camelCase variables and functions in JavaScript. I generally try to name the files somewhat related to the functions that they're containing.

**Comment by thatsnotright** [comment]
This issue is affecting me following the tutorial, I had to modify cli/dist/lib/index.js and remove the capital G on Linux in order to run `yarn redwood generate page home /`

**Comment by thedavidprice** [comment]
Thanks @thatsnotright We hopefully took care of that issue early today — coming in the next package update soon! See #204

**Comment by thedavidprice** [comment]
It seems we’ve settled on **camelCase**.

And reminder to pay extra attention to _never_ using PascalCase as it fails on some systems.

**Comment by cannikin** [comment]
So what did we decide here? Because we currently generate *everything* in PascalCase. Is that changing application-wide?

**Comment by mojombo** [comment]
This isn’t a problem with PascalCase though, right? If I’m not mistaken, this was our code referring to directories with different case, which is bad form no matter what.

**Comment by thedavidprice** [comment]
This discussion goes back to #204 and specifically the first comment:
> the cli directory that contains templates is alternately named Generate and generate in the source. This fails on case-sensitive file systems, but would not fail on case-preserving/case-insensitive file systems.

@mojombo You’re correct, the root issue was the reference with a different case. But that kicked off a conversation about whether or not we need to be concerned about capitalization in naming directories as a best practice (and there’s not much guidance I could find). Rob brought up snake_case as a solution (e.g. probably some reason here why it’s used for “node_modules”).

**tl;dr**
As long as we aren’t creating bugs or issues right now, it doesn’t seem like it matters. So far there have been no reported issues in our App with any case outside the reference issue above.

But our styles _are_ all over the place. So it wouldn’t hurt to try and be consistent going forward, although I don’t suggest we do a full revision to standardize.

Our main Redwood repo has a mix of uses:
- camelCase, e.g. within “cli/" for “dbCommands/"
- hyphens for packages and files, e.g. "create-redwood-app/" and "create-redwood-app.js"

Our App is fairly consistent:
- starter template is mostly lowercase, one word
- “README_APP.md" is an exception (That’s my doing…)
- Prisma migrations use hyphens (not in our control)
- generated directories are all PascalCase

**Comment by thedavidprice** [comment]
¯\_(ツ)_/¯
